Console/retro/Console.cc

```diff
@@ -543,25 +544,17 @@ void Console::Reshape(Rect newBounds)
         std::copy(src, src + std::min(cols, newCols), dst);
     }
     chars.swap(newChars);
-    /*newChars = std::vector<char>(newRows*newCols, ' ');
-    for(short row = 0; row < newRows && row < rows; row++)
-    {
-        char *src = &chars[row * cols];
-        char *dst = &newChars[row * newCols];
-        std::copy(src, src + std::min(cols, newCols), dst);
-    }
-    onscreen.swap(newChars);*/
+
     onscreen = newChars;
 
     rows = newRows;
     cols = newCols;
 
-    if(upshift)
-    {
-        //dirtyRect = Rect { 0, 0, rows, cols };
-        //Update();
-        Draw();
-    }
+
+    dirtyRect = Rect { 0, 0, rows, cols };
+    EraseRect(&newBounds);
+    Update();
```

Comment:
The `Update` call might not do anything useful. It's supposed to only draw cells where `chars` is not equal to `onscreeen`, and it looks like I've not left `onscreen` in a sensible state. Have to think about that again.
Calling `EraseRect` and `Draw` will make things flicker, but it should fix things: `Draw` draws everything and then does `onscreen = chars;`

Reply:
I kept things on the safe side. I used a SE/30 for the moment and the redraw is fast enough, but I played a little with an emulated 68000 machine and the flicker is indeed quite visible.